storage/queue.go

```diff
@@ -400,8 +401,12 @@ func (bq *baseQueue) remove(index int) {
 // Exposed for testing only.
 func (bq *baseQueue) DrainQueue(clock *hlc.Clock) {
 	bq.Lock()
-	defer bq.Unlock()
-	for repl := bq.pop(); repl != nil; repl = bq.pop() {
+	repl := bq.pop()
```

Comment:
Why is that a concern? Even with a single lock acquisition new replicas could be added just before or after the call to DrainQueue. As long as we're not adding new replicas faster than we can process then then DrainQueue will process *at least* one full cycle, which is what we want.